### DISCUSSION TITLE
Correlation coefficients

### DISCUSSION THREAD

**Discussion by gmsilveira** [ROBelgium/MSNoise#297] [General]
I usually compute auto- and cross-correlations using PCC + tf-PWS (Schimmel et al., 2011). And, from the PCC and can easily output the correlation coefficient. I'm now using MSNOISE as I would like to compare the waveform similarity (with PCC+tf_PWS) with the results from MWCS and stretching.

As I am new to MSNOISE, I probably do not see the obvious. But, do we have the output of the correlation coefficient?

Thanks a lot.

**Comment by gmsilveira** [comment]
I just saw a comment saying that this needs to be done using the daily CCFs from within the STACKS directory by correlating them with the reference CCF. Correct?

**Comment by ThomasLecocq** [comment]
Hey Graça,

I'm not familiar (never used, shame on me) PCC & tf-PWS, does that output the dv/v (dt/t) values and a a corrcoef?

The corrcoef is calculated either when using the stretching (it's indeed in the STR folder once done), or a sort of coef/coherence is calculated using the MWCS+DTT method (see Clarke's paper on why this is not really an "error" on the measurements)

**Reply by gmsilveira** [reply]
Hi Thomas,
"PCC & tf-PWS, does that output the dv/v (dt/t) values and a a corrcoef?" 
No, no, this produces only the EGFs. Not dv/v or dt/t. I mean that when I cross-correlate a current EGF with a reference one I can easily output the corrcoef.

"The corrcoef is calculated either when using the stretching (it's indeed in the STR folder once done), or a sort of coef/coherence is calculated using the MWCS+DTT method (see Clarke's paper on why this is not really an "error" on the measurements)"

I only recently realized that we can also compute stretching with MSNOISE (shame on me). Will do it asap, and thanks for the tip on the MCWS+DTT.

Best
Graça

**Reply by ThomasLecocq** [reply]
right, then yes, the best is to interface with the MSNoise API programmatically, and open your resulting CCFs computed from PCC etc. See http://msnoise.org/doc/master/auto_examples/plot_interferogram.html#sphx-glr-auto-examples-plot-interferogram-py to learn how to get the CCFs for your needs. Of course, you can also read the CCF in the STACKS/001_DAYS/* folders directly
